### ISSUE TITLE
Parsing of long floating point numbers causes panic

### ISSUE THREAD

**Issue by MatejKastak** [Geal/nom#1421]
Hello, first of all thank you, nom is a great tool!

I have tried to migrate to a new nom version (7.0.0) and encountered some unexpected behavior.
- parsing of floating point numbers (`float`, `double`) panics for long inputs
- sadly I found no way to handle this error, program is aborted
- are there any tips to avoid such panics?
- recent `6.*` versions does not panic
- migrating to the most recent version is not critical to me, but I wanted to ask/note this behavior for others

After a little digging, I think this is caused by switching to `minimal-lexical`.
> the lexical-core crate used for float parsing has now been replaced with minimal-lexical: the new crate is faster to compile, faster to parse, and has no dependencies

Somewhat related to #1384 

## Prerequisites

- Rust version : `rustc 1.55.0 (c8dfcfe04 2021-09-06)` also tried it with nightly
- nom version : `v7.0.0`
- nom compilation features used: default

## Test case

```rust
mod tests {
    use nom::number::complete::float;
    use nom::IResult;

    fn parser(s: &str) -> IResult<&str, f32> {
        float(s)
    }

    #[test]
    fn float_panic() {
        // 7.0.0 -> panic
        // 6.2.1 -> pass
        assert!(std::matches!(parser("0.00000000000000000087"), Ok(_)));
        assert!(std::matches!(
            parser("0.00000000000000000000000000000000000000000000000000000000000000087"),
            Ok(_)
        ));

    }
}               
```

nom `v7.0.0`
```
---- tests::float_panic stdout ----
thread 'tests::float_panic' panicked at 'attempt to shift left with overflow', /home/anon/.cargo/registry/src/github.com-1ecc6299db9ec823/minimal-lexical-0.1.4/src/lemire.rs:155:5
```

nom `v6.2.1`
```
running 1 test
test tests::float_panic ... ok
```

**Comment by parasyte** [comment]
This should have been closed by #1455

**Comment by Geal** [comment]
Yes. Leaving it open for now, I'd like to get a proper fix in minimal-lexical, because we take a perf hit with that workaround
